src/operations/AuthOperations.ts

```diff
@@ -0,0 +1,192 @@
+import type { WorkOS } from '@workos-inc/node';
+import type { AuthKitCore } from '../core/AuthKitCore.js';
+import type { ConfigurationProvider } from '../core/config/ConfigurationProvider.js';
+import type {
+  AuthResult,
+  BaseTokenClaims,
+  GetAuthorizationUrlOptions,
+  Session,
+} from '../core/session/types.js';
+
+/**
+ * AuthOperations provides high-level authentication operations.
+ *
+ * This class orchestrates between AuthKitCore (pure logic) and WorkOS API.
+ * It doesn't know about requests, responses, or cookies - that's the framework's job.
+ *
+ * Responsibilities:
+ * - Sign out (generate logout URL + clear cookie header)
+ * - Switch organization (refresh with new org + return auth result)
+ * - Refresh session (refresh tokens + return auth result)
+ * - Get authorization URLs (for sign in/sign up flows)
+ */
+export class AuthOperations {
+  constructor(
+    private core: AuthKitCore,
+    private client: WorkOS,
+    private config: ConfigurationProvider,
+  ) {}
+
+  /**
+   * Sign out operation.
+   *
+   * Returns the WorkOS logout URL and a cookie clear header string.
+   * The framework is responsible for applying the header and redirecting.
+   *
+   * @param sessionId - The session ID to terminate
+   * @param options - Optional return URL
+   * @returns Logout URL and clear cookie header
+   */
+  async signOut(
+    sessionId: string,
+    options?: { returnTo?: string },
+  ): Promise<{
+    logoutUrl: string;
+    clearCookieHeader: string;
+  }> {
+    // Generate WorkOS logout URL
+    const logoutUrl = this.client.userManagement.getLogoutUrl({
+      sessionId,
+      returnTo: options?.returnTo,
+    });
+
+    // Build cookie clear header
+    const cookieName = this.config.getValue('cookieName');
+    const clearCookieHeader = `${cookieName}=; Path=/; Max-Age=0; HttpOnly; Secure; SameSite=lax`;
```

Comment:
not a blocker but i think for a general-purpose library, operating on cookie headers seems like the wrong level of abstraction (like suppose a framework exists that has a proper `session` abstraction, we wouldn't want to be setting/clearing cookies directly).